src/modules/common/providers/UserContext.js

```diff
@@ -1,19 +1,65 @@
-import React, { createContext, useState } from 'react'
+import React, { createContext, useEffect, useState } from 'react'
 import PropTypes from 'prop-types'
+import axios from 'axios'
 
 export const UserContext = createContext({})
 
 export const UserProvider = ({ children }) => {
-    const [user, setUser] = useState({
+    const [user, setUserState] = useState({
         firstName: '',
         lastName: '',
         username: '',
         email: '',
         token: ''
     })
 
+    const setUser = ({ firstName, lastName, username, email, token }) => {
+        localStorage.setItem('userToken', token)
+        setUserState({ firstName, lastName, username, email, token })
+    }
+
+    useEffect(() => {
+        async function validateToken() {
+            if (user.token) {
+                try {
+                    const token = localStorage.getItem('userToken')
+                    const response = await axios.get(
+                        'http://localhost:3001/auth/validateToken',
+
+                        {
+                            headers: {
+                                Authorization: `Bearer ${token}`,
+                                'Content-Type': 'application/json'
+                            }
+                        }
+                    )
+
+                    if (response.data.user && setUser(response.data.user)) {
```

Comment:
```suggestion
                    if (response.data.user) {
```
you already set on line below